cro-clib/src/wallet.rs

```diff
@@ -0,0 +1,233 @@
+use chain_core::init::address::RedeemAddress;
+use chain_core::init::network::Network;
+use chain_core::state::account::StakedStateAddress;
+use chain_core::tx::data::address::ExtendedAddr;
+use client_common::MultiSigAddress;
+use client_common::{PrivateKey, PublicKey};
+use client_core::{HDSeed, Mnemonic};
+use secstr::SecUtf8;
+use std::ffi::CStr;
+use std::os::raw::c_char;
+use std::ptr;
+pub type HDWalletPtr = *mut HDWallet;
+pub type AddressPtr = *mut Address;
+
+pub const SUCCESS: i32 = 0;
+pub const FAIL: i32 = -1;
+
+/// network kinds
+#[repr(C)]
+pub enum ApiNetwork {
+    Mainnet,
+    Testnet,
+    Devnet,
+}
+
+/// account types
+#[repr(C)]
+pub enum Account {
+    /// Account for transfer address
+    Transfer = 0,
+    /// Account for staking address
+    Staking = 1,
+    /// Account for viewkey
+    Viewkey = 2,
+}
+
+#[derive(Clone)]
+pub struct HDWallet {
+    pub seed: HDSeed,
+}
+
+#[derive(Clone)]
+pub struct Address {
+    pub privatekey: PrivateKey,
+    pub publickey: PublicKey,
+    pub address: String,
+}
+
+impl From<ApiNetwork> for Network {
+    fn from(src: ApiNetwork) -> Self {
+        match src {
+            ApiNetwork::Mainnet => Network::Mainnet,
+            ApiNetwork::Testnet => Network::Testnet,
+            ApiNetwork::Devnet => Network::Devnet,
+        }
+    }
+}
+
+#[allow(dead_code)]
+unsafe fn get_string(src: *const c_char) -> String {
+    CStr::from_ptr(src).to_string_lossy().into_owned()
+}
+
+#[allow(dead_code)]
+fn copy_string(src: &str, dst: &mut [u8]) {
+    dst[..src.len()].copy_from_slice(&src.as_bytes()[..src.len()]);
+    dst[src.len()] = 0;
+}
+
+/// hdwallet creating using bip44 hdwallet
+#[no_mangle]
+pub unsafe extern "C" fn cro_create_hdwallet(
+    wallet_out: *mut HDWalletPtr,
+    mnemonics: *mut u8,
+    mnemonics_length: i32,
+) -> i32 {
```

Comment:
perhaps you can create a type alias:
```
pub type CroResult = i32;
```

and use that alias instead of i32, so it's then easier to change in the future  (e.g. if it's changed to return a struct with error code + message as in the earlier iteration)

Reply:
changed to CroResult